### PR TITLE
Add support for go-elasticsearch

### PR DESCRIPTION
**Pull request by CodePrometheus**
### Summary

[go-elasticsearch](https://github.com/elastic/go-elasticsearch) is the official Go client for Elasticsearch.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -12,9 +12,7 @@ Release Notes.
 * Support [Segmentio-Kafka](https://github.com/segmentio/kafka-go) MQ.
 * Support http headers collection for Gin
 * Support higher versions of grpc.
-
-#### Chore
```
Comment (mrproliu) [inline CHANGES.md:16]: Why remove this Chore selection?

File: plugins/go-elasticsearchv8/intercepter.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+func (es *ESV8Interceptor) BeforeInvoke(invocation operator.Invocation) error {
+	client := invocation.CallerInstance().(*elasticsearch.BaseClient)
+	var addresses []string
+	for _, u := range client.Transport.(*elastictransport.Client).URLs() {
+		addresses = append(addresses, u.String())
+	}
+	url := strings.Join(addresses, ",")
+	req := invocation.Args()[0].(*http.Request)
+	span, err := tracing.CreateExitSpan("Elasticsearch/"+req.Method, url, func(headerKey, headerValue string) error {
+		req.Header.Add(headerKey, headerValue)
```
Comment (mrproliu) [inline plugins/go-elasticsearchv8/intercepter.go:44]: I think we haven't ability to monitoring the elastic-search or open-search, so the header it's not necessary.

File: plugins/go-elasticsearchv8/intercepter.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+	"github.com/elastic/elastic-transport-go/v8/elastictransport"
+	"github.com/elastic/go-elasticsearch/v8"
+)
+
+type ESV8Interceptor struct {
+}
+
+func (es *ESV8Interceptor) BeforeInvoke(invocation operator.Invocation) error {
+	client := invocation.CallerInstance().(*elasticsearch.BaseClient)
```
Comment (mrproliu) [inline plugins/go-elasticsearchv8/intercepter.go:36]: Is possible we can move the address parsing to the constructor  the client?

File: test/plugins/scenarios/go-elasticsearchv8/config/excepted.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            startTime: nq 0
+            endTime: nq 0
+            componentId: 47
+            isError: false
+            spanType: Exit
+            peer: http://elasticsearch:9200
+            skipAnalysis: false
+            tags:
+              - { key: db.type, value: Elasticsearch }
+              - { key: db.statement, value: not null }
```
Comment (mrproliu) [inline test/plugins/scenarios/go-elasticsearchv8/config/excepted.yml:36]: Could you help adding a demo statement in there?

**Comment by wu-sheng** [comment]
@mrproliu If you want to cut 0.5 release, please move this into 0.6 milestone.

**Comment by mrproliu** [comment]
Sure, I'm still waiting user test. After that finished, I can cut the release.

**Review by mrproliu — APPROVED** [review]

**Pull request merged**